BHoM_Adapter/BHoMAdapter.cs

```diff
@@ -51,21 +51,25 @@ public abstract partial class BHoMAdapter
         /**** Public Adapter Methods                    ****/
         /***************************************************/
 
-        public virtual List<IObject> Push(IEnumerable<IObject> objects, string tag = "", Dictionary<string, object> config = null)
+        public virtual List<IObject> Push(IEnumerable<IObject> objects, string tag = "", Dictionary<string, object> pushConfig = null)
         {
             bool success = true;
 
+            // Get the Push Type from the pushConfig.
             string pushType;
-
             object ptObj;
-            if (config != null && config.TryGetValue("PushType", out ptObj))
+            if (pushConfig != null && pushConfig.TryGetValue("PushType", out ptObj))
                 pushType = ptObj.ToString();
             else
                 pushType = "Replace";
 
-            List<IObject> objectsToPush = objects.Select(x => x is BHoMObject ? ((BHoMObject)x).DeepClone() : x).ToList(); //ToList() necessary for the return collection to function properly for cloned objects
+            // Wrap non-BHoM objects into a Custom BHoMObject to make them work as BHoMObjects.
+            List<IObject> objectsToPush = Modify.WrapNonBHoMObjects(objects, Config, tag, pushConfig).ToList();
 
+            // Clone the objects for immutability in the UI. CloneBeforePush should always be true, except for very specific cases.
+            objectsToPush = Config.CloneBeforePush ? objects.Select(x => x.DeepClone()).ToList() : objects.ToList();
```

Comment:
@IsakNaslundBh the reason for `WrapNonBHoMObjects()` checking the boolean `WrapNonBHoMObjects` inside the method itself is because this way we can freely override the default Adapter setting through the Push config.